python/pyspark/sql/utils.py

```diff
@@ -71,6 +71,15 @@
     pass
 
 
+has_arrow: bool = False
```

Comment:
I think we can use `from pyspark.testing.utils import have_pyarrow`

Reply:
Can we address this comment @xinrong-meng

Reply:
Good point! I think it’s more intuitive in the other direction, having the tests reference sql/utils instead. Let me file a follow-up for easier review, thank you!

Reply:
Good point! Resolved, thank you

Reply:
There will be a circular import if we do that. Let me follow up with a separate PR instead.

Reply:
probably we can import it inside the function/class body